core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/JarResultBuildStep.java

```diff
@@ -260,7 +260,8 @@ public FileVisitResult visitFile(Path file, BasicFileAttributes attrs)
 
             copyCommonContent(runnerZipFs, services, applicationArchivesBuildItem, transformedClasses, generatedClasses,
                     generatedResources, seen);
-            generateManifest(runnerZipFs, classPath.toString(), packageConfig);
+            AppArtifact appArtifact = curateOutcomeBuildItem.getEffectiveModel().getAppArtifact();
+            generateManifest(runnerZipFs, classPath.toString(), packageConfig, appArtifact);
```

Comment:
Thinking more about it, I think we should also take into account `ApplicationInfoBuildItem` if we can inject it there (I put this comment here but it's true for all the other occurrences).

Basically, we would take into account `ApplicationInfoBuildItem` information if it is set and if they are `UNSET_VALUE` (it should be tested separately for `name` and `version`), we would default to the artifactId/version from the pom.

Reply:
:) Made the changes.